test/dynamics/backend/test_dynamics_backend.py

```diff
@@ -127,7 +127,18 @@ def test_measure_nonexistant_subsystem(self):
             pulse.play(pulse.Waveform([0.5, 0.5, 0.5]), pulse.DriveChannel(0))
             pulse.acquire(duration=1, qubit_or_channel=1, register=pulse.MemorySlot(0))
 
-        with self.assertRaisesRegex(QiskitError, "Attempted to measure subsystem 1"):
+        with self.assertRaisesRegex(QiskitError, "Attempted to measure out of bounds subsystem 1."):
+            self.simple_backend.run(schedule)
+
+    def test_measure_trivial_subsystem(self):
+        """Attempt to measure subsystem with dimension 1."""
+
+        with pulse.build() as schedule:
+            pulse.play(pulse.Waveform([0.5, 0.5, 0.5]), pulse.DriveChannel(0))
+            pulse.acquire(duration=1, qubit_or_channel=1, register=pulse.MemorySlot(0))
+
+        self.simple_backend.set_options(subsystem_dims=[2, 1])
```

Comment:
[Q] What situation is the case that specifying explicitly one of subsystem_dims is 1?

Reply:
I wouldn't expect a user to set this themselves, but I wanted to have an explicit test case for this.

Reply:
If users use like 
```python
backend= DynamicsBackend.from_backend(FakeKolkata(), subsystem_list=[1])
```
do they observe the warning?
If so, I'm not sure why this is necessary.

Reply:
The warning will be raised if they do
```
backend= DynamicsBackend.from_backend(FakeKolkata(), subsystem_list=[1])
```
then submit a schedule or circuit that measures qubit 0. I was imagining in this case that we should let users know that they're measuring a system that has no evolution (which I thought could confuse users if no warning were present).